org: antoniocot
service: api-cuentas

provider:
  name: aws
  runtime: python3.12
  memorySize: 1024
  timeout: 30
  iam:
    role: arn:aws:iam::316129865556:role/LabRole
  environment:
    TABLE_NAME: TABLA-CUENTA

functions:
  CrearCuenta:
    handler: CrearCuenta.lambda_handler
    events:
      - http:
          path: /cuentas/crear
          method: post
          cors: true
          integration: lambda

  ListarCuentas:
    handler: ListarCuentas.lambda_handler
    events:
      - http:
          path: /cuentas/listar
          method: post
          cors: true
          integration: lambda

  BuscarCuenta:
    handler: BuscarCuenta.lambda_handler
    events:
      - http:
          path: /cuentas/buscar
          method: post
          cors: true
          integration: lambda

  EliminarCuenta:
    handler: EliminarCuenta.lambda_handler
    events:
      - http:
          path: /cuentas/eliminar
          method: delete
          cors: true
          integration: lambda

  ModificarCuenta:
    handler: ModificarCuenta.lambda_handler
    events:
      - http:
          path: /cuentas/modificar
          method: put
          cors: true
          integration: lambda

  AgregarSaldoCuenta:
    handler: agregarSaldoCuenta.lambda_handler
    events:
      - http: # El problema era aqu√≠; faltaba el array para "events"
          path: /cuentas/agregar-saldo
          method: post
          cors: true
          integration: lambda 

resources:
  Resources:
    CuentasTable:
      Type: AWS::DynamoDB::Table
      Properties:
        TableName: ${self:provider.environment.TABLE_NAME}
        AttributeDefinitions:
          - AttributeName: usuario_id
            AttributeType: S
          - AttributeName: cuenta_id
            AttributeType: S
        KeySchema:
          - AttributeName: usuario_id
            KeyType: HASH
          - AttributeName: cuenta_id
            KeyType: RANGE
        BillingMode: PAY_PER_REQUEST
        PointInTimeRecoverySpecification:
          PointInTimeRecoveryEnabled: true
